Fixes "unknown method router"

rjclardy opened the pull request:
I was getting [this error](https://gist.github.com/rjclardy/4a1bc713df330f4d2841cb63fe9265e6) after adding a `relation` attribute to my schema. It appears that global helpers like `router` aren’t available within view presenter blocks. So this PR changes it so that we’re no longer attempting to call `router` from inside a `with` block.

rjclardy committed:
Fixes "unknown method router"

Global helpers like `router` aren’t available within view presenter
blocks. This changes it so that we’re no longer attempting to call
`router` from inside a `with` block.

bryanp reviewed (changes requested):
If you accept the view as an argument to `with`, it should work as-is:

```ruby
view.component(:modal).with do |view|
  ...
end
```

Without the argument, the block is evaled in context of the view object, which no longer knows about the larger context it's operating within.

rjclardy committed:
Refactor relation editor view logic

Turns out that accepting the view as an argument to `with` allows the
block to be evaluated in the current context such that global helpers
like `router` are available.

rjclardy commented:
Makes sense. Tested it out and it works, thanks!
